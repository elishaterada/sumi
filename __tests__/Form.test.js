import React from 'react';
import renderer from 'react-test-renderer';
import { createSerializer } from 'jest-emotion';
import * as emotion from 'emotion';

import {
  FieldsetClass,
  TextInputClass,
  TextAreaClass,
  LabelClass,
  SelectClass,
  CheckboxClass,
  CheckboxLabelClass,
  RadioClass,
  RadioLabelClass,
} from '../src/components/Form';

expect.addSnapshotSerializer(createSerializer(emotion));

test('renders text input', () => {
  const reactComponent = renderer
    .create(
      <fieldset className={FieldsetClass}>
        <label htmlFor="TextInputClass" className={LabelClass}>Text Input Label</label>
        <input id="TextInputClass" type="text" className={TextInputClass} />
      </fieldset>,
    )
    .toJSON();

  expect(reactComponent).toMatchSnapshot();
});

test('renders textarea', () => {
  const reactComponent = renderer
    .create(
      <fieldset className={FieldsetClass}>
        <label htmlFor="TextAreaClass" className={LabelClass}>Textarea Label</label>
        <textarea id="TextAreaClass" className={TextAreaClass} />
      </fieldset>,
    )
    .toJSON();

  expect(reactComponent).toMatchSnapshot();
});


test('renders select', () => {
  const reactComponent = renderer
    .create(
      <fieldset className={FieldsetClass}>
        <label htmlFor="SelectClass" className={LabelClass}>Select Label</label>
        <select id="SelectClass" className={SelectClass}>
          <option>Please Select</option>
          <option value="item-1">Item 1</option>
          <option value="item-2">Item 2</option>
          <option value="item-3">Item 3</option>
        </select>
      </fieldset>,
    )
    .toJSON();

  expect(reactComponent).toMatchSnapshot();
});

test('renders checkbox', () => {
  const reactComponent = renderer
    .create(
      <fieldset className={FieldsetClass}>
        <input
          id="checkbox-1"
          type="checkbox"
          name="checkbox-group-1"
          value="checkbox-1"
          className={CheckboxClass}
        />
        <label htmlFor="checkbox-1" className={CheckboxLabelClass}>Checkbox 1</label>
        <input
          id="checkbox-2"
          type="checkbox"
          name="checkbox-group-1"
          value="checkbox-2"
          className={CheckboxClass}
        />
        <label htmlFor="checkbox-2" className={CheckboxLabelClass}>Checkbox 2</label>
      </fieldset>,
    )
    .toJSON();

  expect(reactComponent).toMatchSnapshot();
});

test('renders radio', () => {
  const reactComponent = renderer
    .create(
      <fieldset className={FieldsetClass}>
        <input
          id="radio-3"
          type="radio"
          name="radio-group-2"
          value="radio-3"
          className={RadioClass}
        />
        <label htmlFor="radio-3" className={RadioLabelClass}>Radio 1</label>
        <input
          id="radio-4"
          type="radio"
          name="radio-group-2"
          value="radio-4"
          className={RadioClass}
        />
        <label htmlFor="radio-4" className={RadioLabelClass}>Radio 2</label>
      </fieldset>,
    )
    .toJSON();

  expect(reactComponent).toMatchSnapshot();
});
