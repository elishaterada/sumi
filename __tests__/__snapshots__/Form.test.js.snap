// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders checkbox 1`] = `
.emotion-4 {
  margin: 0 0 2rem;
  padding: 0;
  border: 0;
}

.emotion-0 {
  position: absolute;
  outline: none;
  opacity: 0;
}

.emotion-0:checked + label:after {
  opacity: 1;
}

.emotion-0 + label {
  position: relative;
  padding-left: 1.75rem;
  cursor: pointer;
}

.emotion-0 + label:before {
  position: absolute;
  left: 0;
  top: -0.0625rem;
  width: 1.25rem;
  height: 1.25rem;
  display: block;
  content: '';
}

.emotion-0 + label:after {
  position: absolute;
  top: -0.0625rem;
  left: 0;
  width: 1.25rem;
  height: 1.25rem;
  line-height: 1.25rem;
  text-align: center;
  opacity: 0;
  content: '';
}

.emotion-0:disabled + label {
  opacity: 0.5;
  cursor: not-allowed;
}

.emotion-0:hover + label:before,
.emotion-0:focus + label:before {
  background: url('data:image/svg+xml;charset=utf8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2020%2020%22%3E%3Cpath%20d%3D%22M17%200H3C1.3%200%200%201.4%200%203v14c0%201.7%201.3%203%203%203h14c1.7%200%203-1.3%203-3V3c0-1.6-1.3-3-3-3z%22%2F%3E%3Cpath%20d%3D%22M19%2017c0%201.1-.9%202-2%202H3c-1.1%200-2-.9-2-2V3c0-1.1.9-2%202-2h14c1.1%200%202%20.9%202%202v14z%22%20fill%3D%22%23e6e6e6%22%2F%3E%3C%2Fsvg%3E');
}

.emotion-1 {
  margin: 0 1rem 1rem 0;
}

.emotion-1:last-of-type {
  margin: 0 1rem 0 0;
}

.emotion-1:before {
  background: url('data:image/svg+xml;charset=utf8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2020%2020%22%3E%3Cpath%20d%3D%22M17%201H3c-1.1%200-2%20.9-2%202v14c0%201.1.9%202%202%202h14c1.1%200%202-.9%202-2V3c0-1.1-.9-2-2-2z%22%20fill%3D%22%23ffffff%22%2F%3E%3Cpath%20d%3D%22M17%200H3C1.3%200%200%201.4%200%203v14c0%201.7%201.3%203%203%203h14c1.7%200%203-1.3%203-3V3c0-1.6-1.3-3-3-3zm2%2017c0%201.1-.9%202-2%202H3c-1.1%200-2-.9-2-2V3c0-1.1.9-2%202-2h14c1.1%200%202%20.9%202%202v14z%22%20fill%3D%22%23333333%22%20%2F%3E%3C%2Fsvg%3E') 0 center no-repeat;
}

.emotion-1:after {
  background: url('data:image/svg+xml;charset=utf8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2020%2020%22%3E%3Cpath%20d%3D%22M15.3%205.2c-.3-.2-.6-.2-.9%200L9%2012.3%205.7%209.7c-.3-.2-.6-.2-.9%200l-.6.5c-.3.3-.3.6%200%20.8l4.4%203.8c.1.1.2.2.4.2s.4-.1.5-.2l6.4-8.3c.1-.1.1-.5%200-.8l-.6-.5z%22%20fill%3D%22%23ffffff%22%2F%3E%3Cpath%20d%3D%22M17%200H3C1.3%200%200%201.4%200%203v14c0%201.7%201.3%203%203%203h14c1.7%200%203-1.3%203-3V3c0-1.6-1.3-3-3-3zm-1.1%206.5l-6.4%208.3c-.1.1-.3.2-.5.2s-.3-.1-.4-.2L4.2%2011c-.3-.2-.3-.5%200-.8l.6-.5c.3-.2.6-.2.9%200L9%2012.3l5.4-7.1c.3-.2.6-.2.9%200l.6.5c.1.3.1.7%200%20.8z%22%20fill%3D%22%23333333%22%2F%3E%3C%2Fsvg%3E') 0 center no-repeat;
}

<fieldset
  className="emotion-4"
>
  <input
    className="emotion-0"
    id="checkbox-1"
    name="checkbox-group-1"
    type="checkbox"
    value="checkbox-1"
  />
  <label
    className="emotion-1"
    htmlFor="checkbox-1"
  >
    Checkbox 1
  </label>
  <input
    className="emotion-0"
    id="checkbox-2"
    name="checkbox-group-1"
    type="checkbox"
    value="checkbox-2"
  />
  <label
    className="emotion-1"
    htmlFor="checkbox-2"
  >
    Checkbox 2
  </label>
</fieldset>
`;

exports[`renders radio 1`] = `
.emotion-4 {
  margin: 0 0 2rem;
  padding: 0;
  border: 0;
}

.emotion-0 {
  position: absolute;
  outline: none;
  opacity: 0;
}

.emotion-0:checked + label:after {
  opacity: 1;
}

.emotion-0 + label {
  position: relative;
  padding-left: 1.75rem;
  cursor: pointer;
}

.emotion-0 + label:before {
  position: absolute;
  left: 0;
  top: -0.0625rem;
  width: 1.25rem;
  height: 1.25rem;
  display: block;
  content: '';
}

.emotion-0 + label:after {
  position: absolute;
  top: -0.0625rem;
  left: 0;
  width: 1.25rem;
  height: 1.25rem;
  line-height: 1.25rem;
  text-align: center;
  opacity: 0;
  content: '';
}

.emotion-0:disabled + label {
  opacity: 0.5;
  cursor: not-allowed;
}

.emotion-0:hover + label:before,
.emotion-0:focus + label:before {
  background: url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2020%2020%22%3E%3Ccircle%20cx%3D%2210%22%20cy%3D%2210%22%20r%3D%228.9%22%20fill%3D%22%23e6e6e6%22%2F%3E%3Cpath%20fill%3D%22%23333333%22%20d%3D%22M10%2019.9c5.5%200%2010-4.4%209.9-9.9%200-5.4-4.4-9.9-9.9-9.9C4.6.1.1%204.5.1%2010c0%205.4%204.4%209.9%209.9%209.9zm0-18.8c5%200%208.9%204%208.9%208.9s-4%208.9-8.9%208.9-8.9-4-8.9-8.9%204-8.9%208.9-8.9z%22%2F%3E%3C%2Fsvg%3E');
}

.emotion-1 {
  margin: 0 1rem 1rem 0;
}

.emotion-1:last-of-type {
  margin: 0 1rem 0 0;
}

.emotion-1:before {
  background: url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2020%2020%22%3E%3Ccircle%20cx%3D%2210%22%20cy%3D%2210%22%20r%3D%228.9%22%20fill%3D%22%23ffffff%22%2F%3E%3Cpath%20fill%3D%22%23333333%22%20d%3D%22M10%2019.9c5.5%200%2010-4.4%209.9-9.9%200-5.4-4.4-9.9-9.9-9.9C4.6.1.1%204.5.1%2010c0%205.4%204.4%209.9%209.9%209.9zm0-18.8c5%200%208.9%204%208.9%208.9s-4%208.9-8.9%208.9-8.9-4-8.9-8.9%204-8.9%208.9-8.9z%22%2F%3E%3C%2Fsvg%3E') 0 center no-repeat;
}

.emotion-1:after {
  background: url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2020%2020%22%3E%3Ccircle%20cx%3D%2210%22%20cy%3D%2210%22%20r%3D%223.5%22%20fill%3D%22%23ffffff%22%2F%3E%3Cpath%20fill%3D%22%23333333%22%20d%3D%22M10%2019.9c5.5%200%2010-4.4%209.9-9.9%200-5.4-4.4-9.9-9.9-9.9C4.6.1.1%204.5.1%2010c0%205.4%204.4%209.9%209.9%209.9zm0-13.4c1.9%200%203.5%201.6%203.5%203.5s-1.6%203.5-3.5%203.5-3.5-1.6-3.5-3.5S8.1%206.5%2010%206.5z%22%2F%3E%3C%2Fsvg%3E') 0 center no-repeat;
}

<fieldset
  className="emotion-4"
>
  <input
    className="emotion-0"
    id="radio-3"
    name="radio-group-2"
    type="radio"
    value="radio-3"
  />
  <label
    className="emotion-1"
    htmlFor="radio-3"
  >
    Radio 1
  </label>
  <input
    className="emotion-0"
    id="radio-4"
    name="radio-group-2"
    type="radio"
    value="radio-4"
  />
  <label
    className="emotion-1"
    htmlFor="radio-4"
  >
    Radio 2
  </label>
</fieldset>
`;

exports[`renders select 1`] = `
.emotion-2 {
  margin: 0 0 2rem;
  padding: 0;
  border: 0;
}

.emotion-0 {
  display: block;
  margin-bottom: 0.5rem;
  color: #333333;
  font-size: 0.875rem;
}

.emotion-1 {
  margin: 0 0 1.5rem;
  position: relative;
  box-sizing: border-box;
  background: url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2032%2032%22%3E%3Cpath%20fill%3D%22%23333333%22%20d%3D%22M32%209c0%20.5-.2%201-.6%201.4l-14%2014c-.4.4-.9.6-1.4.6s-1-.2-1.4-.6l-14-14C.2%2010%200%209.5%200%209s.2-1%20.6-1.4S1.5%207%202%207h28c.5%200%201%20.2%201.4.6s.6.9.6%201.4z%22%2F%3E%3C%2Fsvg%3E') 95% / 6% no-repeat #ffffff;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  min-width: 15rem;
  max-width: 100%;
  padding: 0.625rem;
  border: 0.0625rem solid #666666;
  border-radius: 0.1875rem;
  font-size: 1rem;
  cursor: pointer;
}

.emotion-1:disabled {
  background: #f2f2f2;
  cursor: not-allowed;
}

<fieldset
  className="emotion-2"
>
  <label
    className="emotion-0"
    htmlFor="SelectClass"
  >
    Select Label
  </label>
  <select
    className="emotion-1"
    id="SelectClass"
  >
    <option>
      Please Select
    </option>
    <option
      value="item-1"
    >
      Item 1
    </option>
    <option
      value="item-2"
    >
      Item 2
    </option>
    <option
      value="item-3"
    >
      Item 3
    </option>
  </select>
</fieldset>
`;

exports[`renders text input 1`] = `
.emotion-2 {
  margin: 0 0 2rem;
  padding: 0;
  border: 0;
}

.emotion-0 {
  display: block;
  margin-bottom: 0.5rem;
  color: #333333;
  font-size: 0.875rem;
}

.emotion-1 {
  margin: 0 0 1.5rem;
  box-sizing: border-box;
  background: #ffffff;
  min-width: 15rem;
  max-width: 100%;
  padding: 0.625rem;
  border: 0.0625rem solid #666666;
  border-radius: 0.1875rem;
  font-size: 1rem;
}

.emotion-1:disabled {
  background: #f2f2f2;
  cursor: not-allowed;
}

<fieldset
  className="emotion-2"
>
  <label
    className="emotion-0"
    htmlFor="TextInputClass"
  >
    Text Input Label
  </label>
  <input
    className="emotion-1"
    id="TextInputClass"
    type="text"
  />
</fieldset>
`;

exports[`renders textarea 1`] = `
.emotion-2 {
  margin: 0 0 2rem;
  padding: 0;
  border: 0;
}

.emotion-0 {
  display: block;
  margin-bottom: 0.5rem;
  color: #333333;
  font-size: 0.875rem;
}

.emotion-1 {
  box-sizing: border-box;
  background: #ffffff;
  min-width: 15rem;
  max-width: 100%;
  height: 6rem;
  padding: 0.625rem;
  border: 0.0625rem solid #666666;
  border-radius: 0.1875rem;
  font-size: 1rem;
}

.emotion-1:disabled {
  background: #f2f2f2;
  cursor: not-allowed;
}

<fieldset
  className="emotion-2"
>
  <label
    className="emotion-0"
    htmlFor="TextAreaClass"
  >
    Textarea Label
  </label>
  <textarea
    className="emotion-1"
    id="TextAreaClass"
  />
</fieldset>
`;

exports[`renders toggle checkbox 1`] = `
.emotion-1 {
  margin: 0 0 2rem;
  padding: 0;
  border: 0;
}

.emotion-0 {
  width: 0.00625rem;
  height: 0.00625rem;
  opacity: 0;
  overflow: hidden;
  position: absolute;
  z-index: -1;
}

.emotion-0:focus + label {
  background-color: #666666;
}

.emotion-0 + label {
  display: block;
  text-indent: -999999rem;
  color: transparent;
  position: relative;
  cursor: pointer;
  outline: none;
  width: 3.75rem;
  height: 1.25rem;
  background-color: #999999;
  border-radius: 2.8125rem;
  -webkit-transition: background 0.25s ease;
  transition: background 0.25s ease;
}

.emotion-0 + label::after {
  content: "";
  display: block;
  position: absolute;
}

.emotion-0 + label::after {
  top: -0.3125rem;
  left: 0;
  width: 2rem;
  height: 2rem;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0.0625rem 0.0625rem 0.625rem rgba(0,0,0,0.15),0 0 0.0625rem rgba(0,0,0,0.15);
  -webkit-transition: all 0.25s ease;
  transition: all 0.25s ease;
}

.emotion-0:checked + label {
  background-color: #333333;
}

.emotion-0:checked + label::before {
  background-color: #333333;
}

.emotion-0:checked + label::after {
  left: 100%;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.emotion-0:disabled + label {
  opacity: 0.5;
  cursor: default;
}

<fieldset
  className="emotion-1"
>
  <input
    className="emotion-0"
    id="checkbox-5"
    type="checkbox"
    value="checkbox-5"
  />
  <label
    htmlFor="checkbox-5"
  >
    Checkbox
  </label>
</fieldset>
`;
